dotnet/private/actions/assembly_common.bzl

```diff
@@ -218,3 +243,184 @@ def emit_assembly_common(
     )
 
     return new_library
+
+def _emit_assembly_common_fsharp(
+        kind,
+        dotnet,
+        name,
+        srcs,
+        deps = None,
+        out = None,
+        resources = None,
+        executable = True,
+        defines = None,
+        unsafe = False,
+        data = None,
+        keyfile = None,
+        subdir = "./",
+        target_framework = "",
+        nowarn = None,
+        langversion = "latest",
+        version = (0, 0, 0, 0, "")):
+    if name == "" and out == None:
+        fail("either name or out must be set")
+
+    if not out:
+        result = dotnet.declare_file(dotnet, path = subdir + name)
+    else:
+        result = dotnet.declare_file(dotnet, path = subdir + out)
+
+    if dotnet.debug:
+        pdbext = ".mdb" if kind == "mono" else ".pdb"
+        pdb = dotnet.declare_file(dotnet, path = result.basename + pdbext, sibling = result)
+    else:
+        pdb = None
+
+    direct_inputs = []
+    args = dotnet.actions.args()
+
+    # General command lines parameters
+    args.add(result.path, format = "/out:%s")
+    if executable:
+        target = "exe"
+    else:
+        target = "library"
+    args.add(target, format = "/target:%s")
+
+    args.add("/checked-")
+    args.add("/utf8output")
+    args.add("/deterministic+")
+    args.add("/nologo")
+    args.add("/highentropyva+")
+    args.add("/noframework")
+    args.add("/nowin32manifest")
+    args.add("/targetprofile:netcore")
+    args.add("/simpleresolution")
+    args.add("/nocopyfsharpcore")
+
+    # Debug parameters
+    if pdb:
+        args.add("-debug:full")
```

Comment:
`full` isn't useful for .net core anymore, all tools should be using `portable` (if they want standalone PDBs), or `embedded` (if they want PDBs embedded in the output assembly itself).

Reply:
or you could do like the C# targets a bit above and put `debug:full` behind a check for `mono`, but I wouldn't even suggest that, as mono can read the `portable` pdb format.

Reply:
I would consider supporting F# via separate toolchain. The version of rules_dotnet I'm working on is moving towards deeper support of Bazel platforms & toolchains.